mldsa/polyvec.h

```diff
@@ -221,11 +236,18 @@ void polyveck_shiftl(polyveck *v);
  * Name:        polyveck_ntt
  *
  * Description: Forward NTT of all polynomials in vector of length MLDSA_K.
- *Output coefficients can be up to 16*MLDSA_Q larger than input coefficients.
+ *              Coefficients can grow by 8*MLDSA_Q in absolute value.
  *
  * Arguments:   - polyveck *v: pointer to input/output vector
  **************************************************/
-void polyveck_ntt(polyveck *v);
+void polyveck_ntt(polyveck *v)
+__contract__(
+  requires(memory_no_alias(v, sizeof(polyveck)))
+  requires(forall(k0, 0, MLDSA_K, array_abs_bound(v->vec[k0].coeffs, 0, MLDSA_N, MLDSA_Q)))
+  assigns(memory_slice(v, sizeof(polyveck)))
+  ensures(forall(k1, 0, MLDSA_K, array_abs_bound(v->vec[k1].coeffs, 0, MLDSA_N, MLD_NTT_BOUND)))
+);
+
```

Comment:
Looks good to me! I had got to a similar state with the proof I started on this

```
__contract__(
  requires(memory_no_alias(v, sizeof(polyvecl)))
  requires(forall(i, 0, MLDSA_L,
    array_abs_bound(v->vec[i].coeffs, 0, MLDSA_N, MLDSA_Q)))
  assigns(object_whole(v))
  ensures(forall(i, 0, MLDSA_L,
    array_abs_bound(v->vec[i].coeffs, 0, MLDSA_N, MLD_NTT_BOUND)))
);
```
But I had abandoned due to issue 8617